packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx

```diff
@@ -17,24 +17,28 @@ import {
   TextField,
   Typography,
 } from '@mui/material'
-import { parse, ParseError } from 'papaparse'
-import { RJSFSchema } from '@rjsf/utils'
+import { GenericObjectType, RJSFSchema } from '@rjsf/utils'
 import TextWidget from '../SchemaDrivenAnnotationEditor/widget/TextWidget'
 import { DateTimeWidget } from '../SchemaDrivenAnnotationEditor/widget/DateTimeWidget'
 import { SelectWidget } from '../SchemaDrivenAnnotationEditor/widget/SelectWidget'
 import { BooleanWidget } from '../SchemaDrivenAnnotationEditor/widget/BooleanWidget'
 import { getTransformErrors } from '../SchemaDrivenAnnotationEditor/AnnotationEditorUtils'
 import ErrorListTemplate from '../SchemaDrivenAnnotationEditor/template/ErrorListTemplate'
+import useParseCsv from './useParseCsv'
+import { ParseError } from 'papaparse'
 
-export type JSONArrayEditorProps = {
-  value?: string[]
+const DEFAULT_ARRAY_ITEM_DEFINITION: JSONSchema7Definition = { type: 'string' }
+
+export type JSONArrayEditorProps<T = unknown> = {
+  value?: T[]
   arrayItemDefinition?: JSONSchema7Definition
-  onChange: (newValue: string[]) => void
-  onSubmit: (formData: string[]) => void
+  onChange: (newValue: T[]) => void
+  onSubmit: (formData: T[]) => void
+  formRef?: React.Ref<RJSFForm<T, RJSFSchema, GenericObjectType>>
```

Comment:
Instead of using React.forwardRef, add a `formRef` prop because React.forwardRef doesn't play nicely with TypeScript generics